build_tools/ci/matmul_aiert_stability/run_matmul_test.sh

```diff
@@ -0,0 +1,565 @@
+  --lhs_rhs_type "bf16" \
+  --acc_type "f32" \
+  --m "128" --n "256" --k "512" \
+  --do_transpose_rhs "1"
+
+###################################################################
+
+run_matmul_test \
+    --name_prefix "mm2" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "308"  --k "9728" --n "2432"
+
+run_matmul_test \
+    --name_prefix "mm3" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "308"  --k "2432" --n "2432"
+
+run_matmul_test \
+     --name_prefix "mm4" \
+     --lhs_rhs_type "bf16" \
+     --acc_type "f32" \
+     --m "308"  --k "2432" --n "7296"
+
+run_matmul_test \
+     --name_prefix "mm5" \
+     --lhs_rhs_type "bf16" \
+     --acc_type "f32" \
+     --m "8192" --k "2432" --n "9728"
+
+run_matmul_test \
+    --name_prefix "mm6" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "308"  --k "2432" --n "9728"
+
+run_matmul_test \
+    --name_prefix "mm7" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "8192" --k "2432" --n "2432"
+
+run_matmul_test \
+     --name_prefix "mm8" \
+     --lhs_rhs_type "bf16" \
+     --acc_type "f32" \
+     --m "8192" --k "9728" --n "2432"
+
+run_matmul_test \
+    --name_prefix "mm9" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "8192" --k "2432" --n "7296"
+
+###################################################################
+# ObjectFifo Matmul tests
+###################################################################
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "32" --k "32" --n "32"
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "64" --k "32" --n "128"
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "128" --k "32" --n "64"
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "128" --k "32" --n "128"
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "256" --k "32" --n "256"
+
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "128" --k "256" --n "128"
+
+run_matmul_test \
+    --name_prefix "medium" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "i32" \
+    --acc_type "i32" \
+    --m "1024" --k "1024" --n "1024"
+
+run_matmul_test \
```

Comment:
are you sure there is value in adding so many of these, I just worry if we make some expected change, it will be a lot of burden to fix the tests.

Reply:
I mean NA issues are "P0" right? Debugging them is a ...nightmare... on AIE. Actually on the very commit that added these tests (~30 minutes ago) there was an NA fail). So locally right now I actually added dumping the actual register values.... but I'm not gonna put that into the checked in files. In their current form they actually shouldn't change unless you change `amdaie-...` passes. So that falls to me currently and I'm fine maintaining them.

Reply:
lo and behold it caught something 

```diff
**** Generating matmul .aiert.log file for mm_small_i32_i32_m32_n32_k32 ****
1874c1874
< cdo-driver:     Address: 0x0000000000320E6C  Data is: 0x16010219 
---
> cdo-driver:     Address: 0x0000000000320E6C  Data is: 0x16410219 
1877c1877
< cdo-driver:     Address: 0x0000000000320E78  Data is: 0x00011641 
---
> cdo-driver:     Address: 0x0000000000320E78  Data is: 0x00011601 
2819c2819
< cdo-driver:     Address: 0x0000000002220E6C  Data is: 0x16010219 
---
> cdo-driver:     Address: 0x0000000002220E6C  Data is: 0x16410219 
2822c2822
< cdo-driver:     Address: 0x0000000002220E78  Data is: 0x00011641 
---
> cdo-driver:     Address: 0x0000000002220E78  Data is: 0x00011601 
3764c3764
< cdo-driver:     Address: 0x0000000002320E6C  Data is: 0x16010219 
---
> cdo-driver:     Address: 0x0000000002320E6C  Data is: 0x16410219 
3767c3767
< cdo-driver:     Address: 0x0000000002320E78  Data is: 0x00011641 
---
> cdo-driver:     Address: 0x0000000002320E78  Data is: 0x00011601 
```

Reply:
ok sounds good! so lets say we change something with tiling, this check test wont change then?

Reply:
Adding my comment here to centralize:

Isn't the output going to change all the time, meaning we would have to update these files all the time? Instead, we could for example just test the routing connection parts and/or reduce the number of tests (I think quite bunch of those will have the exact same routing).

Reply:
after much deliberation and politicking we decided to add the script but decide later (soon) how to use it (nightly? commit to commit? etc).